### PR TITLE
DayViewDecorator Revamp

### PR DESCRIPTION
**Pull request by dandc87**
This isn't strictly about DayViewDecorators, but thats mostly what it is.

Breaking Changes:
* DayViewDecorators are now date agnostic, i.e. every date that should be decorated will be decorated the same. This allows for more optimized decorating at the expense of flexibility. 
    * The biggest change is that in place of `setText()` there is now `addSpan()`.
    * `setBackgroundUnselected()` has changed to `setBackgroundDrawable()` and is always drawn instead of just in unselected states.
    * `setBackground()` has changed to `setSelectionDrawable()` which is just better naming
    * `decorate()` should be called only once per invalidation. The results are applied to all dates where `shouldDecorate()` 

New: 
* Add new Span `DotSpan` which will draw a dot under text

Samples:
* Added first day of the week to several samples
* Improved decorator sample

### CODE REVIEW

**Comment by dandc87** [comment]
@ireneduke @Jogan do you want to review this?

File: sample/build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ android {
 dependencies {
     compile 'com.android.support:appcompat-v7:22.1.1'
     compile 'com.android.support:recyclerview-v7:22.1.1'
-    //compile project(':library')
-    compile 'com.prolificinteractive:material-calendarview:0.3.2'
+    compile project(':library')
+    //compile 'com.prolificinteractive:material-calendarview:0.3.2'
```
Comment (ireneduke) [inline sample/build.gradle]: Why do you have this line commented out? Should it be removed?

Comment (dandc87) [inline sample/build.gradle]: I keep the line around for convenience when I test new releases to maven. The last release I forgot to swap it back to using the project line.

Comment (ireneduke) [inline sample/build.gradle]: Can you add a comment about why it's here so it's not confusing for anyone in that case?

Comment (dandc87) [inline sample/build.gradle]: Documented

**Comment by ireneduke** [comment]
:+1:

**Pull request merged**